diff --git a/00_LCP2116/LCT1_r1_b/main.cpp b/00_LCP2116/LCT1_r1_b/main.cpp
--- a/00_LCP2116/LCT1_r1_b/main.cpp
+++ b/00_LCP2116/LCT1_r1_b/main.cpp
@@ -1170,20 +1170,23 @@ void modbusProceed(int *regs, int addr_, int reg_qty_, int Value_)
 			case 2:
 			regs[i] = convertFloat(params.RMS_A,0);
+			break;
 			
 			case 3:
 			regs[i] = convertFloat(params.RMS_A,1);
 			break;
 			
 			case 4:
 			regs[i] = convertFloat(params.RMS_B,0);
+			break;
 			
 			case 5:
 			regs[i] = convertFloat(params.RMS_B,1);
 			break;
 
 			case 6:
 			regs[i] = convertFloat(params.RMS_C,0);
+			break;
 			
 			case 7:
 			regs[i] = convertFloat(params.RMS_C,1);
diff --git a/00_LCP2116/LCT2_r8_p330/main.cpp b/00_LCP2116/LCT2_r8_p330/main.cpp
--- a/00_LCP2116/LCT2_r8_p330/main.cpp
+++ b/00_LCP2116/LCT2_r8_p330/main.cpp
@@ -2728,9 +2728,9 @@ void modbusProceed(int16_t *regs, uint16_t addr_, uint16_t reg_qty_, int16_t Va
 			{
 				// Check if the corresponding bit in AI_TypeByte is set (1)
 				if (AI_TypeByte & 0b00000100)
-				regs[i] = convertFloat((cts.solenoidCurrOff_2_RAW - b[SOL_OFF_2]) * k[SOL_OFF_1], 0);
+				regs[i] = convertFloat((cts.solenoidCurrOff_2_RAW - b[SOL_OFF_2]) * k[SOL_OFF_2], 0);
 				else
-				regs[i] = convertFloat((solenoidCurrOff_2[CALC] - b[SOL_OFF_2]) * k[SOL_OFF_1], 0);
+				regs[i] = convertFloat((solenoidCurrOff_2[CALC] - b[SOL_OFF_2]) * k[SOL_OFF_2], 0);
 			}
 			break;
